src/modules/str/libkvistr.cpp

```diff
@@ -1721,41 +1721,109 @@ static bool str_kvs_fnc_split(KviKvsModuleFunctionCall * c)
 	KVSM_PARAMETERS_END(c)
 
 	if(c->params()->count() < 4)
-	    iMaxItems = -1;
+		iMaxItems = -1;
 
 	KviKvsArray * a = new KviKvsArray();
 	c->returnValue()->setArray(a);
 
-	if(iMaxItems == 0)
-		return true;
-	if(iMaxItems == 1)
+	if(szSep.isEmpty())
 	{
-		a->append(new KviKvsVariant{szStr});
+		a->set(0, new KviKvsVariant(szStr));
 		return true;
 	}
 
-	bool bWild = szFla.contains('w', Qt::CaseInsensitive);
-	bool bContainsR = szFla.contains('r', Qt::CaseInsensitive);
+	if(iMaxItems == 0)
+		return true;
+
+	bool bWild = szFla.indexOf('w', 0, Qt::CaseInsensitive) != -1;
+	bool bContainsR = szFla.indexOf('r', 0, Qt::CaseInsensitive) != -1;
+	bool bCaseSensitive = szFla.indexOf('s', 0, Qt::CaseInsensitive) != -1;
+	bool bNoEmpty = szFla.indexOf('n', 0, Qt::CaseInsensitive) != -1;
+
+	int id = 0;
 
-	Qt::SplitBehavior splitBehavior = szFla.contains('n', Qt::CaseInsensitive) ? Qt::SkipEmptyParts : Qt::KeepEmptyParts;
-	Qt::CaseSensitivity sensitivity = szFla.contains('s', Qt::CaseInsensitive) ? Qt::CaseSensitive : Qt::CaseInsensitive;
+	int iMatch = 0;
+	int iStrLen = szStr.length();
+	int iBegin = 0;
+
+	if(bContainsR || bWild)
+	{
+		KviRegExp re(szSep, bCaseSensitive ? KviRegExp::CaseSensitive : KviRegExp::CaseInsensitive, bWild ? KviRegExp::Wildcard : KviRegExp::RegExp);
+
+		while((iMatch != -1) && (iMatch < iStrLen) && ((id < (iMaxItems - 1)) || (iMaxItems < 0)))
+		{
+			iMatch = re.indexIn(szStr, iBegin);
+			if(iMatch != -1)
+			{
+				int len = re.matchedLength();
+				if((len == 0) && (iBegin == iMatch))
+					iMatch++; // safety measure for empty string matching
 
-	QVector<QStringRef> list;
-	if(bWild || bContainsR)
-		list = szStr.splitRef(QRegExp{szSep, sensitivity, bWild ? QRegExp::Wildcard : QRegExp::RegExp2}, splitBehavior);
+				QString tmp = szStr.mid(iBegin, iMatch - iBegin);
+				if(bNoEmpty)
+				{
+					if(!tmp.isEmpty())
+					{
+						a->set(id, new KviKvsVariant(tmp));
+						id++;
+					}
+				}
+				else
+				{
+					a->set(id, new KviKvsVariant(tmp));
+					id++;
+				}
+
+				iMatch += len;
+				iBegin = iMatch;
+			}
+		}
+	}
 	else
-		list = szStr.splitRef(szSep, splitBehavior, sensitivity);
+	{
```

Comment:
Lots of changes in this function. I don't see in the PR description what this is for, so I don't know how to review this change

Reply:
Sorry, I missed it in the PR description. 

IceN9ne's rewrote $str.split() in 2017 making use of QStringRef / QString::splitRef(), but they unfortunately doesn't exists anymore in Qt6.
I reverted a part of this commit: https://github.com/kvirc/KVIrc/commit/9e8a9d9e0d38993e145dddab64b42114efc4d3f6#diff-2c8a6d9d89531ab98f04a8dce7a8d6670d4661adac0b78edd4b5a38941890befR1717
The previous version of $str.split() wasn't making use of those Qt features and could be ported without further changes